lib/web_ui/lib/src/engine/semantics/label_and_value.dart

```diff
@@ -14,12 +16,413 @@ import 'semantics.dart';
 /// respect the `aria-label` without a [DomText] node. Crawlers typically do not
 /// need this information, as they primarily scan visible text, which is
 /// communicated in semantics as leaf text and heading nodes.
-enum LeafLabelRepresentation {
+enum LabelRepresentation {
   /// Represents the label as an `aria-label` attribute.
+  ///
+  /// This representation is the most efficient as all it does is pass a string
+  /// to the browser that does not incur any DOM costs.
+  ///
+  /// The drawback of this representation is that it is not compatible with most
+  /// web crawlers, and for some ARIA roles (including the implicit "generic"
+  /// role) JAWS on Windows. However, this role is still the most common, as it
+  /// applies to all container nodes, and many ARIA roles (e.g. checkboxes,
+  /// radios, scrollables, sliders).
   ariaLabel,
 
   /// Represents the label as a [DomText] node.
+  ///
+  /// This is the second fastest way to represent a label in the DOM. It has a
+  /// small cost because the browser lays out the text (in addition to Flutter
+  /// having already done it).
+  ///
+  /// This representation is compatible with most web crawlers, and it is the
+  /// best option for certain ARIA roles, such as buttons, links, and headings.
   domText,
+
+  /// Represents the label as a sized span.
+  ///
+  /// This representation is the costliest as it uses an extra element that
+  /// need to be laid out to compute the right size. It is compatible with most
+  /// web crawlers, and it is the best options for certain ARIA roles, such as
+  /// the implicit "generic" role used for plain text (not headings).
+  sizedSpan;
+
+  /// Creates the behavior for this label representation.
+  LabelRepresentationBehavior createBehavior(PrimaryRoleManager owner) {
+    return switch (this) {
+      LabelRepresentation.ariaLabel => AriaLabelRepresentation._(owner),
+      LabelRepresentation.domText => DomTextRepresentation._(owner),
+      LabelRepresentation.sizedSpan => SizedSpanRepresentation._(owner),
```

Comment:
This is also possible:
```suggestion
      ariaLabel => AriaLabelRepresentation._(owner),
      domText => DomTextRepresentation._(owner),
      sizedSpan => SizedSpanRepresentation._(owner),
```